cluster-autoscaler/FAQ.md

```diff
@@ -297,140 +308,143 @@ kubectl annotate node <nodename> cluster-autoscaler.kubernetes.io/scale-down-dis
 
 ### How does CA deal with unready nodes in version >=0.5.0 ?
 
-From 0.5 CA (K8S 1.6) continues the work even if some (up to 33% or not greater than 3, configurable via flag) percentage of nodes
-is unavailable. Once there are more unready nodes in the cluster, CA pauses all operations until the situation
-improves. If there are fewer unready nodes but they are concentrated in a particular node group
-then this node group may be excluded from scale-ups.
-Prior to 0.5, CA stopped all operations when a single node became unready.
+From 0.5 CA (K8S 1.6) continues to work even if some (up to 33% or not greater than 3, configurable by ``--max-total-unready-percentage` and `--ok-total-unready-count` flags) percentage of nodes
+is unavailable. Once there are more unready nodes in the cluster, CA stops all operations until the situation improves. If there are fewer unready nodes, but they are concentrated in a particular node group, then this node group may be excluded from future scale-ups.
 
 ### How fast is Cluster Autoscaler?
 
-Scale up (if it is reasonable) is executed up to 10 seconds after some pod is marked as unschedulable.
-Scale down is executed (by default) 10 min (or later) after a node becomes unneeded.
+By default, scale-up is considered up to 10 seconds after pod is marked as unschedulable, and scale-down 10 minutes after a node becomes unneeded. There are multiple flags which can be used to configure them. Assuming default settings, [SLOs described here apply.](#what-are-the-service-level-objectives-for-cluster-autoscaler)
 
 ### How fast is HPA when combined with CA?
 
-By default, Pod CPU usage is scraped by kubelets every 10 sec, and CPU usage is obtained from kubelets by Heapster every 1 min.
-HPA checks cpu load metrics in Heapster every 30 sec, and CA looks for unschedulable pods every 10 sec. So the max reaction
-time, measured from the time CPU spikes in the pods to the time CA asks the cloud provider for a new node is 2 min. On average
-it should be around 1 min.
-The amount of time the cloud provider needs to start a new node and boot it up is measured in minutes. On GCE/GKE it is around 1.5-2 min -
-however this depends on the data center location and machine type.
-Then it may take up to 30 sec to register the node in the Kubernetes master and finalize all of the necessary network settings.
+By default, pods' CPU usage is scraped by kubelet every 10 sec, and it is obtained from kubelet
+by Heapster every 1 min. HPA checks CPU load metrics in Heapster every 30 sec,
+and CA looks for unschedulable pods every 10 sec. So the maximum reaction time,
+measured from the time CPU spikes in the pods to the time CA asks the cloud
+provider for a new node, is about 2 min. On average, it should be around 1 min.
+The amount of time needed to start a new node and boot it up depends on cloud
+provider and is measured in minutes. Then it may take up to 30 seconds to register the node
```

Comment:
"up to 30s" to register node in k8s? I'd like to see that :P
I'm aware it has been here before, but we may as well change it to some reasonable number.

Reply:
IIUC this refers only to time it takes kubelet to register after it was deployed and so isn't completely unrealistic. On the other hand, this is probably irrelevant to the end user from cluster autoscaling point of view. So are details of HPA metrics pipeline. Can we maybe consider removing it? @mwielgus WDYT?